controller/src/main/java/org/jboss/as/controller/ModelControllerImpl.java

```diff
@@ -1212,27 +1192,8 @@ ManagementModelImpl cloneRootResource() {
         }
 
         ManagementModelImpl cloneCapabilityRegistry() {
-            ManagementResourceRegistration mrr;
-            Resource currentResource;
-            CapabilityRegistryImpl currentCaps;
-            if (published) {
-                // This is the first clone since this was published. Use the current stuff as the basis
-                // to ensure that the clone is based on the latest even if we are not the latest.
-                ManagementModelImpl currentPublished = ModelControllerImpl.this.managementModel.get();
-                mrr = currentPublished.resourceRegistration;
-                currentResource = currentPublished.rootResource;
-                currentCaps = currentPublished.capabilityRegistry;
-            } else {
-                // We've already been cloned, which means the thread calling this has the controller lock
-                // and our stuff hasn't been superceded by another thread. So use our stuff
-                mrr = resourceRegistration;
-                currentResource = rootResource;
-                currentCaps = capabilityRegistry;
-            }
-            CapabilityRegistryImpl clone = currentCaps.copy();
-            ManagementModelImpl result = new ManagementModelImpl(mrr, currentResource, clone);
-            ControllerLogger.MGMT_OP_LOGGER.tracef("cloned to %s to create %s and %s", currentCaps, clone, result);
-            return result;
+            //todo why is this even needed, return this; should be fine.
+            return new ManagementModelImpl(resourceRegistration, rootResource, capabilityRegistry);
```

Comment:
Please resolve this TODO. If cloneCapabilityRegistry then essentially becomes a no-op, the method should be removed.